src/cascadia/TerminalControl/HwndTerminal.cpp

```diff
@@ -102,25 +109,23 @@ try
             }
             break;
         case WM_RBUTTONDOWN:
-            if (publicTerminal->_terminal && publicTerminal->_terminal->IsSelectionActive())
+            try
             {
-                try
+                if (publicTerminal->_terminal)
                 {
-                    Terminal::TextCopyData bufferData;
+                    const auto lock = publicTerminal->_terminal->LockForWriting();
+                    if (publicTerminal->_terminal->IsSelectionActive())
                     {
-                        const auto lock = publicTerminal->_terminal->LockForWriting();
-                        bufferData = publicTerminal->_terminal->RetrieveSelectedTextFromBuffer(false, true, true);
+                        const auto bufferData = publicTerminal->_terminal->RetrieveSelectedTextFromBuffer(false, true, true);
+                        LOG_IF_FAILED(publicTerminal->_CopyTextToSystemClipboard(bufferData.plainText, bufferData.html, bufferData.rtf));
+                        publicTerminal->_ClearSelection();

```

Comment:
This fixes a bug which is noticeable in debug builds. `_ClearSelection` requires the lock being held, but in the old code it didn't do that.